feature/experimental/policy/policy_vrf_selection/otg_tests/protocol_dscp_rules_for_vrf_selection_test/protocol_dscp_rules_for_vrf_selection_test.go

```diff
@@ -260,55 +268,81 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 }
 
 // getIPinIPFlow returns an IPv4inIPv4 *ondatra.Flow with provided DSCP value for a given set of endpoints.
-func getIPinIPFlow(args *testArgs, dstEndPoint ondatra.Endpoint, flowName string, dscp uint8) *ondatra.Flow {
-
-	ethHeader := ondatra.NewEthernetHeader()
-	outerIPHeader := ondatra.NewIPv4Header().WithDSCP(dscp)
-	innerIPHeader := ondatra.NewIPv4Header()
-	innerIPHeader.WithSrcAddress("198.51.100.1")
-	innerIPHeader.DstAddressRange().WithMin("203.0.113.1").WithCount(10000).WithStep("0.0.0.1")
-	flow := args.ate.Traffic().NewFlow(flowName)
-	flow.WithHeaders(ethHeader, outerIPHeader, innerIPHeader)
-	flow.WithSrcEndpoints(args.srcEndPoint)
-	flow.WithDstEndpoints(dstEndPoint)
-	flow.WithFrameRateFPS(100)
-	flow.WithFrameSize(1024)
+func getIPinIPFlow(args *testArgs, src attrs.Attributes, dst attrs.Attributes, flowName string, dscp int32) gosnappi.Flow {
+
+	flow := gosnappi.NewFlow().SetName(flowName)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{src.Name + "." + args.iptype}).SetRxNames([]string{dst.Name + "." + args.iptype})
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(src.MAC)
+	if args.iptype == "ipv4" {
+		outerIPHeader := flow.Packet().Add().Ipv4()
+		outerIPHeader.Src().SetValue(src.IPv4)
+		outerIPHeader.Dst().SetValue(dst.IPv4)
+		if dscp != 0 {
+			outerIPHeader.Priority().Dscp().Phb().SetValue(dscp)
+		}
+	}
+	if args.iptype == "ipv6" {
```

Comment:
It looks like the old version of "getIPinIPFlow" assumed we were testing IPv4, and that in practice the surrounding test only called it with args.iptype == "ipv4".

In the refactored "getIPinIPFlow", you add support for the args.iptype == "ipv6" case, but it looks like it's incomplete as innerIPHeader still uses IPv4 and we don't set dscp in IPv6 (I don't know if these are needed in the "ipv6" case)

I'd suggest either leaving it as only supporting "ipv4", or complete adding support for "ipv6" in getIPinIPFlow.

Reply:
Removed for ipv6 as dscp is not set for IPv6, made it aligned with ATE test